gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -78,6 +84,16 @@
   private final ImmutableMap<String, String> metadata;
   private final boolean checksumsEnabled;
 
+  // Helper delegate for turning IOExceptions from API calls into higher-level semantics.
+  private ApiErrorExtractor errorExtractor = ApiErrorExtractor.INSTANCE;
+  // Determine if a given IOException is due to rate-limiting.
+  private RetryDeterminer<IOException> rateLimitedRetryDeterminer =
+      RetryDeterminer.createRateLimitedRetryDeterminer(errorExtractor);
+  // Object to use to perform sleep operations
+  private Sleeper sleeper = Sleeper.DEFAULT;
+  // BackOff objects are per-request, use this to make new ones.
+  private BackOffFactory backOffFactory = BackOffFactory.DEFAULT;
+
```

Comment:
If you initialize these fields with the same values all the time, then you can just inline them in the call site. Or even better, you can create `ResilientOperation.retry` method that uses default values.

Reply:
Done. Left rateLimitedRetryDeterminer to be there because it needs more code to initialize and can be shared btw multiple methods.